src/spec.ts

```diff
@@ -79,10 +79,12 @@ export interface GenericUnitSpec<M, E extends Encoding> extends BaseSpec {
 
 export type UnitSpec = GenericUnitSpec<Mark, Encoding>;
 
+export type LayeredUnitSpec = GenericUnitSpec<AnyMark, Encoding>;
+
 
 export type FacetedUnitSpec = GenericUnitSpec<AnyMark, EncodingWithFacet>;
 
-export interface GenericLayerSpec<M, E> extends BaseSpec {
+export interface GenericLayerSpec<U> extends BaseSpec {
```

Comment:
I just make another commit for `U extends GenericUnitSpec`.  (Let's squash after we are happy with other things.)